widget.py

```diff
@@ -0,0 +1,309 @@
+
+    def set_bins(bins):
+        """
+        Ustawia mi początkową wartość bins dla wszystkich
+        histogrmaów
+        """
+        Bttn_bins.__bins = bins
+
+    def change_bins(self, event):
+        """
+        Method for keyboard handling.
+        """
+        # Changing number of bins; Number is a power of 2. Max val. 512.
+
+        if self.label == '+':
+            if Bttn_bins.__bins < 512:
+                Bttn_bins.__bins *= 2
+                for his in self.histogram_list:
+                    his.set_bins(Bttn_bins.__bins)
+                    his.update()
+                    his.draw_green_space()
+        elif self.label == '-':
+            if Bttn_bins.__bins > 2:
+                Bttn_bins.__bins /= 2
+                for his in self.histogram_list:
+                    his.set_bins(Bttn_bins.__bins)
+                    his.update()
+                    his.draw_green_space()
+        self.fig.canvas.draw()
+
+
+class Radio(RadioButtons):
+    """
+    class used in hdfsee
+    image - image created by functions imshow()
+    fig - object of the Figuge class used to redraw
+    cmap - color map used to change color map in image
+    """
+    def __init__(self, fig, ax, labels, cmap, image):
+        self.image = image
+        self.fig = fig
+        self.cmap = cmap
+        # Initialize parent constructor.
+        super(Radio, self).__init__(ax=ax, labels=labels, active=0)
+        # On click reaction.
+        self.on_clicked(self.colorfunc)
+
+    def colorfunc(self, label):
+        """
+        Calls itself after switching colour.
+        Updates the image.
+        """
+        # change name cmap
+        self.cmap = label
+        # set cmap
+        self.image.set_cmap(cmap=self.cmap)
+        # Redraw the current figure.
+        self.fig.canvas.draw()
+
+    def set_image(self, image):
+        """
+        set a new object returned by imshow
+        """
+        self.image = image
+
+    def get_cmap(self):
+        """
+        returned last cmap
+        """
+        return self.cmap
+
+
+class My_slider(Slider):
+    """
+    class used in hdfsee
+    image - image created by functions imshow()
+    fig - object of the Figuge class used to redraw
+    vmax and vmin - is used to change contrast
+    """
+    def __init__(self, image, fig, ax, label, vmin, vmax):
+        self.image = image
+        self.fig = fig
+        self.vmax = vmax
+        self.vmin = vmin
+        # Initialize parent constructor.
+        super(My_slider, self).__init__(ax, label, vmin, 600, valinit=255)
+        super(My_slider, self).on_changed(self.on_check)  # On click reaction.
+
+    def on_check(self, event):
+        self.vmax = event
+        self.image.set_clim(vmax=self.vmax)
+        self.fig.canvas.draw()
+
+    def set_image(self, image):
+        self.image = image
+
+    def get_vmax(self):
+        return self.vmax
+
+    def get_vmin(self):
+        return self.vmin
+
+
+class Bttn(Button):
```

Comment:
abreviation

Reply:
Addressed in: e73a1996e75cfceb523bf223bbad6228e6e19c81